### PR TITLE
(MODULES-3880) Ignore Package Exit Codes

### PR DESCRIPTION
**Pull request by ferventcoder**
This might need a ticket @jpogran / @puppetlabs/windows 

A few maintenance fixes that I've been wanting to do prior to the release, the bigger of which is to add the ignore package exit codes.

### CODE REVIEW

**Comment by ferventcoder** [comment]
This passed adhoc testing.

File: lib/puppet/provider/package/chocolatey.rb [hunk 1]
```diff
@@ -84,12 +84,16 @@ def install
...
 
-    args << @resource[:install_options]
-
     if @resource[:source]
       args << '-source' << @resource[:source]
     end
 
+    args << @resource[:install_options]
+
+    if Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon.choco_version) >= Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon::MINIMUM_SUPPORTED_CHOCO_VERSION_EXIT_CODES)
```
Comment (jpogran) [inline lib/puppet/provider/package/chocolatey.rb]: Nit: can these be made variables? Super long line makes its hard to read the important part

File: lib/puppet/provider/package/chocolatey.rb [hunk 2]
```diff
@@ -106,14 +110,19 @@ def uninstall
...
+    choco_version = Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon.choco_version)
+    if !choco_exe || choco_version >= Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon::MINIMUM_SUPPORTED_CHOCO_UNINSTALL_SOURCE)
       if @resource[:source]
         args << '-source' << @resource[:source]
       end
     end
 
+    args << @resource[:uninstall_options]
+
+    if Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon.choco_version) >= Gem::Version.new(PuppetX::Chocolatey::ChocolateyCommon::MINIMUM_SUPPORTED_CHOCO_VERSION_EXIT_CODES)
```
Comment (jpogran) [inline lib/puppet/provider/package/chocolatey.rb]: Same here. Super long line and readability

**Comment by jpogran** [comment]
👍 Can ignore my nit comments if you want

**Pull request merged**